sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1285,7 +1333,12 @@ def _df_getitem_unicode_idx_impl(self, idx):
 
         return gen_df_getitem_bool_series_idx_impl(self, idx)
 
-    ty_checker.raise_exc(idx, 'str', 'idx')
+    if isinstance(idx, types.Array) and isinstance(idx.dtype, types.Boolean):
```

Comment:
I would propose to move it at the top, just after we checked that self is of DataFrameType, because this check should really intercept wrong types (for which we do not want compilation to happen). Otherwise we might compile (or at least try to compile) some specialization we do not want to compile (e.g. if some of the arg types can be type casted to another one).